server/recipeAPI/IRecipeAPI.ts

```diff
@@ -0,0 +1,30 @@
+import IRecipe from "../serverAPI/model/recipe/IRecipe";
+
+export default interface IRecipeAPI {
+     /**
+     * Retrieves collection of partially filled Recipe items that satisfy searching parameters.
+     * 
+     * @param parameters parameters used for searching.
+     * - query - required parameter that defines the name of the Recipe item (partial names are accepted).
+     * - number - optional parameter that defines max number of the results to be returned.
+     * - page - optional parameter that defines page number for pagination.
+     * - intolerence - optional parameter that defines the type of intolerences to be taken in consideration during searching.
+     * 
+     * @throws NoParameterFound exception when required parameters weren't found.
+     *
+     * @returns Promise filled with a collection of Partial<IRecipe> objects.
```

Comment:
Spacing between param throw and returns is different from in SpoonacularAPI for example, same below